src/compiler/types.ts

```diff
@@ -3711,11 +3718,14 @@ export type NamedExportBindings =
 export interface ImportClause extends NamedDeclaration {
     readonly kind: SyntaxKind.ImportClause;
     readonly parent: ImportDeclaration | JSDocImportTag;
-    readonly isTypeOnly: boolean;
+    /** @deprecated */ readonly isTypeOnly: boolean;
+    readonly phaseModifier: undefined | ImportPhaseModifier;
     readonly name?: Identifier; // Default binding
     readonly namedBindings?: NamedImportBindings;
 }
 
+export type ImportPhaseModifier = SyntaxKind.TypeKeyword | SyntaxKind.DeferKeyword;

```

Comment:
Types ending in `Modifier` generally refer to an instantiation of the `ModifierToken` type (or are part of the `Modifier` union of those types). I'd prefer this be renamed to `ImportPhaseModifierSyntaxKind`, much like we do for `ModifierSyntaxKind`, `KeywordTypeSyntaxKind`, etc.